### PR TITLE
First attempt to write some privacy considerations

### PR DESCRIPTION
**Pull request by mirjak**
This ended up in more text that I thought. I'm not certain if we need all of it. Further I'm unsure about the concrete wording in many cases, so any comments are more than welcome!

### CODE REVIEW

**Comment by mirjak** [comment]
This is supposed to address issue #177

**Comment by mirjak** [comment]
I do mentioned name resolution, however, leakage though DNS might rather be a DNS problem and not necessarily an issue that needs to be considered in taps.

Sharing cache state is a good point. I vaguely remember that we already address that somewhere else (could also be in the implementation draft as this is rather an implementation issue). But maybe we can add a pointer...?

**Comment by britram** [comment]
Leakage though DNS may need to be addressed by TAPS, as with the current sockets API, the application has complete control about when DNS resolution is done (and, at least as long as things like DoH/DoT are predominantly implemented by nonsystem resolver libraries, *how*). TAPS as envisioned gives the system more control over resolution.

**Comment by britram** [comment]
(IIRC the architecture does explicitly address and mention the shared cache issue; adding a line to refer to that in privacy considerations might be nice but isn't IMO necessary.)

File: draft-ietf-taps-interface.md [hunk 1]
```diff
@@ -2540,6 +2540,41 @@ This document describes a generic API for interacting with a transport services
 Part of this API includes configuration details for transport security protocols, as discussed
 in {{security-parameters}}. It does not recommend use (or disuse) of specific
 algorithms or protocols. Any API-compatible transport security protocol should work in a TAPS system.
+Security consideration for these protocols should be discussed in the respective specifications.
+
+The desribed API is used to exchange information between an application and the transport system. While
+it is not necessarily expected that both systems are implemented by the same authority, it is expected
+that the transport system implementation is either provided as library that is selected by the application
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md]: as /a/ library

File: draft-ietf-taps-interface.md [hunk 2]
```diff
@@ -2540,6 +2540,41 @@ This document describes a generic API for interacting with a transport services
...
+from a trusted party, or that it is part of the operating system that the application also relies on for
+other tasks.
+
+In any case the TAPS API is an internal interface that is used to change information locally between two systems.
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md]: /In any case/In either case/ ? 
- and add comma?

File: draft-ietf-taps-interface.md [hunk 3]
```diff
@@ -2540,6 +2540,41 @@ This document describes a generic API for interacting with a transport services
...
+However, as the transport system is responsible for network communication, it is in the position to
+potentially share any information provided by the application with the network or another communication peer. 
+Most of the information provided over the TAPS API are useful to configure and select protocols and paths
+and are not necessarily privacy sensitive. Still, there is some information that could be privacy sensitve as
+it might reveal usage characteristics and habits of the user of an application.
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md]: /as it/because this/

File: draft-ietf-taps-interface.md [hunk 4]
```diff
@@ -2540,6 +2540,41 @@ This document describes a generic API for interacting with a transport services
...
+other tasks.
+
+In any case the TAPS API is an internal interface that is used to change information locally between two systems.
+However, as the transport system is responsible for network communication, it is in the position to
+potentially share any information provided by the application with the network or another communication peer. 
+Most of the information provided over the TAPS API are useful to configure and select protocols and paths
+and are not necessarily privacy sensitive. Still, there is some information that could be privacy sensitve as
+it might reveal usage characteristics and habits of the user of an application. 
+
+It should first be noted that of course any communication over a network reveals usage characteristics, as all
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md]: /It should first be noted/
-Could be deleted top read easier?

File: draft-ietf-taps-interface.md [hunk 5]
```diff
@@ -2540,6 +2540,41 @@ This document describes a generic API for interacting with a transport services
...
+packets as well as their timing and size are part of the network-visible wire image {{?RFC8546}}. However, of
+course the selection of a protocol and its configuration impacts which information is visible, potentially in
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md]: /of course/ 
- could be deleted?

File: draft-ietf-taps-interface.md [hunk 6]
```diff
@@ -2540,6 +2540,41 @@ This document describes a generic API for interacting with a transport services
...
+clear text, to which other enties. In most cases information that is provided for protocol and path selection
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md]: /to which other entities/
- seem to read wrong.. is this: and the other entities that can access this, or something else?

File: draft-ietf-taps-interface.md [hunk 7]
```diff
@@ -2540,6 +2540,41 @@ This document describes a generic API for interacting with a transport services
...
+should not directly translate in information that is visible on the path. But there might be specific configuration
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md]: /in information/to information/?
/that is visible on the path/... /that can be observed by netwoprkl devices on the path/?

File: draft-ietf-taps-interface.md [hunk 8]
```diff
@@ -2540,6 +2540,41 @@ This document describes a generic API for interacting with a transport services
...
+This is for example the case when the transport system also excecutes name resolution; or when support mechanisms such as
+TURN or ICE are used to establish connectivity; or if protocols or paths are raised; or if a path fails and 
+fallback or re-establishment is supported in the transport system. 
+
+These communication activities are not different from what is used today, however, the goal of a TAPS transport system is to support
+such mechanisms as a generic service within the transport layer. This enables applications to more dynamically
+benefit from innovations and new protocols in the transport system but at the same time may reduce transparency of the 
+underlying communication actions to the application itself. The TAPS API is designed such that protocol and path selection
+can be limited to a small and controlled set if required by the application for functional or security purposes. Further
+TAPS implementations should provide an interface to pull information about which protocol and path is currently in use as
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md]: /Further,/
"pull" ... probably a better word is needed... is this "provide"

File: draft-ietf-taps-interface.md [hunk 9]
```diff
@@ -2540,6 +2540,41 @@ This document describes a generic API for interacting with a transport services
...
+well as provide logging about communication events of each connection.
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md]: insert /the/ before /communications/

**Comment by mirjak** [comment]
Gorry, I addressed all your nits/comments.

**Pull request merged**